1단계 - 엔티티 매핑

minseokLim opened the pull request:
처음에 클론 받고 좀 당황했습니다;;
리팩토링해야할 부분도 너무 많아보이고 엔티티도 이상하고... 이걸 다 고쳐야하나 생각을 하다 그걸 2, 3단계에서 한다는 걸 알게되었습니다^^;;; 그래서 1단계에서는 가볍게 학습테스트만을 요구한다고 생각하고 다른 건 최대한 건드리지 않기 위해 노력했습니다.

잘 부탁드립니다! :)

minseokLim committed:
chore: 'checkstyle' 플러그인 추가

minseokLim committed:
feat: JPA 학습테스트

nokchax commented on src/main/java/qna/Application.java:
```diff
@@ -2,7 +2,9 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing;
 
+@EnableJpaAuditing
```
👍

nokchax commented on src/main/java/qna/domain/Answer.java:
```diff
@@ -3,14 +3,38 @@
-    private Long questionId;
     private String contents;
+
+    @CreatedDate
+    private LocalDateTime createdAt;
     private boolean deleted = false;
+    private Long questionId;
+
+    @LastModifiedDate
+    private LocalDateTime updatedAt;
```
createdAt, updatedAt은 여러 곳에서 재사용하는 코드인데요, 이런 반복적인 걸 해결할 수 없을까요?

minseokLim replied:
추상클래스를 만들어서 MappedSuperClass 어노테이션을 사용해서 상속받으면 된다는 걸 알고는 있는데, 앞서 말씀드렸듯이 클론 받고 많이 당황해서; 어디까지 뜯어고쳐야하고 어디까지 1단계에서 원하는 건지 범위를 알 수가 없어서 진행하지 않았습니다^^; 해야한다고 생각하시면 진행하도록 하겠습니다.

nokchax replied:
객체간의 연관관계 매핑이 아닌 엔티티 매핑이기 때문에 진행하셔도 될 것 같습니다 :)

nokchax commented on src/main/java/qna/domain/Answer.java:
```diff
@@ -3,14 +3,38 @@
 public class Answer {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
-    private Long writerId;
-    private Long questionId;
     private String contents;
+
+    @CreatedDate
+    private LocalDateTime createdAt;
     private boolean deleted = false;
+    private Long questionId;
+
+    @LastModifiedDate
+    private LocalDateTime updatedAt;
+    private Long writerId;
```
여기에 선언된 것 만으로 요구사항에 나와있는 ddl과 같은 ddl이 생성될까요?
다른 엔티티도 마찬가진데요, 테스트 하실때 출력되는 ddl을 확인해보세요 :)

minseokLim replied:
아이고 죄송합니다ㅠㅠ 출근 전 급하게 하는 바람에 정말 기본적인건데 체크하질 못했네요. 수정하도록 하겠습니다.

nokchax commented on src/main/java/qna/domain/DeleteHistory.java:
```diff
@@ -3,15 +3,25 @@
-    private ContentType contentType;
     private Long contentId;
-    private Long deletedById;
+    private ContentType contentType;
```
`@Enumerated`에 대해 한번 찾아보세요 :)

minseokLim replied:
아.. 이것도 varchar인데 지금은 ordinal로 박히겠군요... 제 불찰입니다ㅠㅠ

nokchax commented on src/test/java/qna/domain/AnswerTest.java:
```diff
@@ -1,6 +1,71 @@
+    @Autowired
+    private AnswerRepository answerRepository;
+
+    @BeforeEach
+    void setUp() {
+        saved = answerRepository.save(new Answer(UserTest.JAVAJIGI, QuestionTest.Q1, "너무 어렵게 생각하지 마라."));
+    }
```
setup에 대해 한번 생각해보셨으면 해서 링크 공유드립니다!
https://jojoldu.tistory.com/611

minseokLim replied:
퇴근 후 확인해보도록 하겠습니다. 감사합니다! :)

nokchax replied:
오늘 테스트에 관해 좋은글을 하나 발견해서 추가적으로 공유드려요 :)
https://blog.kingbbode.com/52

nokchax reviewed (changes requested):
안녕하세요, 민석님! 2주차 JPA 과제 리뷰어 김광현입니다. 🙇‍♂️

엔티티 매핑 과제 전반적으로 잘 진행해주셨어요!
다만 필드와 컬럼을 매핑할때 다양한 어노테이션을 사용하는데,
빠진 것들이 있어 요구사항에 적혀있는 ddl을 만족하지 못하고 있습니다.
확인후 재요청 부탁드립니다 :)

minseokLim committed:
refactor: PR 피드백 반영

1. 주어진 DDL 조건을 모두 충족하도록 엔티티 변경
2. BaseTimeEntity 상속을 통한 생성일과 수정일에 대한 중복 코드 제거
3. 테스트 코드 리팩토링

nokchax commented on src/main/java/qna/domain/BaseTimeEntity.java:
```diff
@@ -0,0 +1,22 @@
+
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
+public abstract class BaseTimeEntity {
```
👍

nokchax reviewed (approved):
안녕하세요, 민석님! 🙇‍♂️ 
피드백 잘 반영해 주셨습니다.
깔끔히 리팩토링 하셔서, 특별히 코멘트 드릴게 없네요 💯 
머지하겠습니다, 다음 단계 진행해주세요!
